website/blog/2021-01/00/_index.md

```diff
@@ -0,0 +1,87 @@
+- `Safety Controller` unfreezes the MCM automatically once situation is resolved to normal. A `freeze` label is applied on `MachineDeployment`/`MachineSet` to enforce the freeze condition.
+
+## Evolution of MCM from In-Tree to Out-of-Tree (OOT)
+MCM supports declarative management of machines in a K8s Cluster on various cloud providers like AWS, Azure, GCP, AliCloud, OpenStack, Metal-stack, Packet, KubeVirt, VMWare, Yandex. It can, of course, be easily extended to support other cloud providers.
+
+Going ahead having the implementation of the Machine Controller Manager supporting too many cloud providers would be too much upkeep from both a development and a maintenance point of view. Which is why, the `Machine Controller` component of MCM has been moved to Out-of-Tree design where `Machine Controller` for respective cloud provider runs as an independent executable; even though typically packaged under the same deployment.
+
+<img title="Figure 2: Out-Of-Tree Machine Controller Manager" src="images/01.png" style="width:90%; height:auto" />
+<figcaption style="text-align:center;margin-top: 0px;margin-bottom: 30px;font-size: 90%;">Figure 2: Out-Of-Tree (OOT) Machine Controller Manager</figcaption>
+
+This OOT Machine Controller will implement a common interface to manage the VMs on the respective cloud provider. Now, while `Machine Controller` deals with the `Machine` objects, Machine Controller Manager (MCM) deals with higher level objects such as `MachineSet` and `MachineDeployment` objects.
+
+## Who uses MCM?
+**[Gardener](http://gardener.cloud)**
+
+MCM is originally developed and employed by a K8s Control Plane as a Service called Gardener. However, the MCM’s design is elegant enough to be employed when managing the machines of any independent K8s clusters, without having to necessarily associate it with Gardener.
+
+**[Sky UK Limited](http://sky.com)**
+
+Sky UK Limited is a British broadcaster and telecommunications company in UK who recently migrated their Kubernetes nodes from Ansible to Machine Controller Manager. Check out [this](https://youtu.be/yF4wq7GAeEM) video from YouTube [Gardener Project](https://www.youtube.com/channel/UCwUhwKFREV8Su0gwAJQX7tw) Channel where Anthony Comtios, Principle Engineer working at Sky talks about MCM in action with Sky's tech stack.
+
+
+## Conclusion
+Machine Controller Manager is so far the best automation for machine management for K8s Clusters. And the best part is that it is open sourced. It gives everyone a scope of both employment and enhancement at ease.
+
+Whether you want to know more about Machine Controller Manager or see a potential scope of the same for your solutions, then visit the GitHub page [machine-controller-manager](https://github.com/gardener/machine-controller-manager). We are so excited to see what you achieve with Machine Controller Manager.
```

Comment:
Whether you want to know more about Machine Controller Manager or find out about a similar scope for your solutions,